### PR TITLE
Fix ToSignificant() output on very small numbers

### LINKED ISSUE #13: ToSignificant misbehavor on small numbers
Calling `ToSignificant` on a price which has more zeros behind the decimal point than requested significant digits returns 0

Tested with:
```go
func TestInvertPrice(t *testing.T) {
	alvn, err := entities.NewToken(constants.Mainnet, common.HexToAddress("0x2488f8ce316cFaa9D9094C87fFf02E4552aC2dbD"), 9, "alvn", "alvn")
	if err != nil {
		t.Error(err)
	}
	busd, err := entities.NewToken(constants.Mainnet, common.HexToAddress("0xe9e7cea3dedca5984780bafc599bd69add087d56"), 18, "busd", "busd")
	if err != nil {
		t.Error(err)
	}
	amountA, ok := new(big.Int).SetString("298093822684209213803885", 10)
	if !ok {
		t.Error("Failed to parse amount")
	}
	tokenAmountA, err := entities.NewTokenAmount(alvn, amountA)
	if err != nil {
		t.Error(err)
	}
	amountB, ok := new(big.Int).SetString("365229067249208740553426", 10)
	if !ok {
		t.Error("Failed to parse amount")
	}
	tokenAmountB, err := entities.NewTokenAmount(busd, amountB)
	if err != nil {
		t.Error(err)
	}
	pair, err := entities.NewPair(tokenAmountA, tokenAmountB)
	if err != nil {
		t.Error(err)
	}
	route, err := entities.NewRoute([]*entities.Pair{pair}, busd, alvn)
	if err != nil {
		t.Error(err)
	}
	amountIn, err := entities.NewTokenAmount(busd, ethutils.ToWei(1, 18))
	if err != nil {
		t.Error(err)
	}
	trade, err := entities.NewTrade(route, amountIn, constants.ExactInput)
	if err != nil {
		t.Fatal(err)
	}
	t.Log(trade.Route.MidPrice.Invert().ToSignificant(2))
}
```

### PR DESCRIPTION
**Pull request by jon4hz**
fixes #13

### CODE REVIEW

File: entities/fraction.go [hunk 1]
```diff
@@ -120,16 +120,32 @@ func (f *Fraction) Divide(other *Fraction) *Fraction {
...
 	d := decimal.NewFromBigInt(f.Numerator, 0).Div(decimal.NewFromBigInt(f.Denominator, 0))
+	if d.LessThan(decimal.New(1, 0)) {
+		significantDigits += countZerosAfterDecimalPoint(d.String())
+	}
+	f.opts.Apply(number.WithRoundingPrecision(int(significantDigits)))
 	if v, err := number.DecimalRound(d, f.opts); err == nil {
 		d = v
 	}
-
+	fmt.Println(d.String())
```
Comment (miraclesu-to) [inline entities/fraction.go:132]: could you help to remove the debug code?

File: entities/fraction.go [hunk 2]
```diff
@@ -120,16 +120,32 @@ func (f *Fraction) Divide(other *Fraction) *Fraction {
...
 	return number.DecimalFormat(d, f.opts)
 }
 
+func countZerosAfterDecimalPoint(d string) uint {
+	grp := strings.Split(d, ".")
+	if len(grp) != 2 {
```
Comment (miraclesu-to) [inline entities/fraction.go:138]: could you replace the magic number `2` with a more interesting const variable?
```suggestion
	if len(grp) != xx {
```

**Review by miraclesu-to — CHANGES_REQUESTED** [review]
Hi @jon4hz, thank you for your contribution, could you modify the code by the comments?
Thanks.

File: entities/fraction.go (hunk 1 above)
Comment (jon4hz) [reply entities/fraction.go]: ah dammit, thought I removed everything.. anyway. It's removed now

File: entities/fraction.go (hunk 2 above)
Comment (jon4hz) [reply entities/fraction.go]: done

**Review by miraclesu — APPROVED** [review]

**Pull request merged**